Unused variable in OptionParser.cpp

**Describe the bug**
I tried to build GEOPM with icpx and I expected it to complete instead it errored on an unused varaible in OptionParser.cpp

**GEOPM version**
Output from `geopmread --version`
2.0.0+dev178g5cf3b31b

**Expected behavior**
A clean build.

**Actual behavior**
An error on unused variables.

**Error messages produced**
`src/OptionParser.cpp:80:13: error: variable 'idx' set but not used [-Werror,-Wunused-but-set-variable]
        int idx = 0;
            ^
src/OptionParser.cpp:80:13: error: variable 'idx' set but not used [-Werror,-Wunused-but-set-variable]
        int idx = 0;`

--- a/src/OptionParser.cpp
+++ b/src/OptionParser.cpp
@@ -77,13 +77,11 @@ namespace geopm
         std::vector<struct option> long_options;
         std::string short_options;
 
-        int idx = 0;
         for (const auto &conf : m_bool_opts) {
             std::string name = conf.first;
             const struct m_opt_parse_s<bool> &opt = conf.second;
             short_options += opt.short_form;
             long_options.push_back({opt.long_form.c_str(), no_argument, NULL, opt.short_form});
-            ++idx;
         }
         for (const auto &conf : m_str_opts) {
             std::string name = conf.first;

